README.md

```diff
@@ -84,149 +94,106 @@ const UserProfile = ({ userId }) => (
+  refresh: Load<A>,
+  // Returns the current state of the resource
+  getState: () => A,
+  // A function that takes the next state or a function that receives the current state and returns the next state.
+  setState: (A | A => A) => void,
+  // Allows for subscribing to resource state changes. Basically a wrapper around store.subscribe.
+  subscribe: (() => void) => void,
+  // A react hook that allows you to use a resource entry's state in the same way that you would use React's useState
+  useEntryState: <A>(...args: Array<any>) => [A, (A | A => A) => void]
 };
 ```
 
-This hook is very powerful. Let's walk through what happens when it is used:
+### `createSingleEntryResource`
 
-1. If there is no data for this resource OR no entry in the cache for the arguments (See [invalidateAfter](https://github.com/chadwatson/react-remote-resource/blob/master/README.md#createremoteresource)) OR the cache is stale (See [invalidateAfter](https://github.com/chadwatson/react-remote-resource/blob/master/README.md#createremoteresource)), then the `load` function of the `categoriesResource` and `productsResource` will be invoked and promises thrown.
-2. If either of the promises reject, the closest `RemoteResourceBoundary` will handle the error. If both promises resolve, the `categoriesById` and `products` data will be available to use (as the first item in the tuple).
-3. You can set the state using the second item in the tuple. Resource state changes, unlike component based `useState`, will persist in memory. If a component unmounts and remounts the state will be the same as when you left it.
+An opinionated version of `createResource` that assumes there is only one entry in the resource state. The getter function simply returns the last data that was fetched. The setter function simply sets the resource state to the data that was fetched. And the predicate function simply checks if the resource state is not `undefined`. Basically the loader function will only be called once if it successfully resolves.
 
-### `useResourceActions`
+```javascript
+const myResource = createSingleEntryResource(authToken =>
+  fetch(`/api/about_me?auth_token=${authToken}`)
+);
+```
 
-A React hook that takes a resource and an optional array of arguments and returns an object literal with the following methods:
+### `createKeyedResource`
 
-- `set`: A function that takes either the new state to set for this entry or a function that takes the current state of the entry and returns what should be set as the new state for the entry.
+An opinionated version of `createResource` that stores retrieved data in an object literal, allowing you to create a key for each entry. The getter function returns the data that is associated with key that your key creating function returns. The setter function sets the resource state to the data that was fetched. And the predicate function simply checks if the resource state is not `undefined`.
```

Comment:
Again, clearer. I'm still not understanding what `(pre set)` means. And I'm thinking we should just take out `opinionated`. And I think I need to change the predicate so that it only checks if it is not `undefined` instead of if it's truthy.

Reply:
What about this?

```markdown
Creates a resource that organizes its entries into an object literal. It takes a key setter function and a loader function that fetches your data. The key setter function derives the entry key from the same arguments that are supplied to the loader function. Once an entry is fetched it will not be refetched.
```

Reply:
👍!